src/strict/core/signal_engine.py

```
@@ -18,6 +18,66 @@ def __init__(self) -> None:
         """Initialize the SignalEngine."""
         pass
 
+    @staticmethod
+    def generate_signal(config: SignalConfig) -> list[float]:
```

Ruff (0.14.14)

[error] Undefined name `SignalConfig`

(F821)